ext/intl/formatter/formatter_main.c

```diff
@@ -104,6 +110,64 @@ PHP_METHOD( NumberFormatter, __construct )
 }
 /* }}} */
 
+/* {{{ proto void NumberFormatter::__wakeup()
+ * NumberFormatter object unserializer.
+ */
+PHP_METHOD( NumberFormatter, __wakeup )
+{
+		const char* locale = intl_locale_get_default();
+		size_t      pattern_len = 0;
+    	char*       pattern = NULL;
+    	zend_long   style = 0;
+    	UChar*      spattern     = NULL;
+    	int32_t     spattern_len = 0;
+		zval        rv;
+		zval        *z_locale, *z_style, *z_pattern;
+		zend_string *str;
+
+		return_value = getThis();
+
+    	z_locale = zend_read_property(Z_OBJCE_P(return_value), return_value, "locale", sizeof("locale") - 1, 0, &rv);
+		if (z_locale != NULL) {
+			if (z_locale && Z_TYPE_P(z_locale) == IS_STRING) {
+				str = zval_get_string(z_locale);
+				locale = ZSTR_VAL(str);
+				zend_string_release(str);
+            }
+		}
+
+		z_style = zend_read_property(Z_OBJCE_P(return_value), return_value, "style", sizeof("style") - 1, 0, &rv);
+		if (z_style != NULL) {
+			style = zval_get_long(z_style);
+		}
+
+		z_pattern = zend_read_property(Z_OBJCE_P(return_value), return_value, "pattern", sizeof("pattern") - 1, 0, &rv);
+		if (z_pattern != NULL) {
+			str = zval_get_string(z_pattern);
+			pattern = ZSTR_VAL(str);
+			pattern_len = ZSTR_LEN(str);
+			zend_string_release(str);
```

Comment:
Same here, this also releases a string that may later be used. Instead you should simply keep the whole zend_string and release it at the end.